### PR TITLE
Add the preview button for better UX

### PR DESCRIPTION
**Pull request by Aritra8438**
# Description
This PR adds the `preview` button so the user can preview the markdown without saving it in the store.

Fixes: #7637

https://github.com/omegaup/omegaup/assets/64671908/8ccd2b39-49e9-4a65-8d19-b9982ff52d8d

# Comment

- The changes will be previewed in the markdown viewer after clicking the `Preview` button.
- The changes will be previewed in the markdown and saved to the store after clicking the `Save` button.

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
Hi @carlosabcs, PTAL!

**Comment by heduenas** [comment]
@Aritra8438 Would it be a lot of work or hurt performance to automatically update the preview as the user types in the textbox?

**Comment by Aritra8438** [comment]
Hi @heduenas, Sure, I think it won't be a problem.
I tried to follow the previous CDP work.

It won't take long to enable that feature. But should I do the same for stores?

Doing so for stores will update the store too frequently.

**Comment by carlosabcs** [comment]
> Hi @heduenas, Sure, I think it won't be a problem. I tried to follow the previous CDP work.
> 
> It won't take long to enable that feature. But should I do the same for store?
> 
> Doing so for store will update the store too frequently with possibly large amount of data.

IMHO, just auto-preview the markdown and save the data in the store with the CTA button.

Also, can we add an alert or something indicating that the text was saved?

**Comment by Aritra8438** [comment]
Hi @carlosabcs, made the necessary changes. PTAL!


> Also, can we add an alert or something indicating that the text was saved?

I have created a separate issue for this, once the creator page is implemented, I will solve this. 
#7643

File: frontend/templates/en.lang [hunk 1]
```diff
@@ -1028,6 +1028,7 @@ problemCreatorGroup = "Group"
 problemCreatorGroupName = "Group Name"
 problemCreatorGroups = "Groups"
 problemCreatorLoadProblem = "Load Problem"
+problemCreatorMarkdownPreviewInitialRender = "#Preview\n"
```
Comment (carlosabcs) [inline frontend/templates/en.lang:1031]:
```suggestion
problemCreatorMarkdownPreviewInitialRender = "# Preview\n"
```

File: frontend/templates/es.lang [hunk 2]
```diff
@@ -1028,6 +1028,7 @@ problemCreatorGroup = "Grupo"
 problemCreatorGroupName = "Nombre del grupo"
 problemCreatorGroups = "Grupos"
 problemCreatorLoadProblem = "Cargar Problema"
+problemCreatorMarkdownPreviewInitialRender = "#Avance\n"
```
Comment (carlosabcs) [inline frontend/templates/es.lang:1031]:
```suggestion
problemCreatorMarkdownPreviewInitialRender = "# Previsualización\n"
```

File: frontend/templates/pt.lang [hunk 3]
```diff
@@ -1028,6 +1028,7 @@ problemCreatorGroup = "Grupo"
 problemCreatorGroupName = "Nome do grupo"
 problemCreatorGroups = "Grupos"
 problemCreatorLoadProblem = "Problema de carga"
+problemCreatorMarkdownPreviewInitialRender = "#Visualizar\n"
```
Comment (carlosabcs) [inline frontend/templates/pt.lang:1031]:
```suggestion
problemCreatorMarkdownPreviewInitialRender = "# Pré-visualização\n"
```

**Review by carlosabcs — APPROVED** [review]
Just fix those small nits

**Comment by Aritra8438** [comment]
@heduenas, PTAL!

**Review by heduenas — APPROVED** [review]

**Pull request merged**